src/IROperator.cpp

```diff
@@ -1421,6 +1421,135 @@ Expr fast_cos(const Expr &x_full) {
     return fast_sin_cos(x_full, false);
 }
 
+// A vectorizable atan and atan2 implementation.
+// Based on the ideas presented in https://mazzo.li/posts/vectorized-atan2.html.
+Expr fast_atan_approximation(const Expr &x_full, ApproximationPrecision precision, bool between_m1_and_p1) {
+    const float pi_over_two = 1.57079632679489661923f;
+    Expr x;
+    // if x > 1 -> atan(x) = Pi/2 - atan(1/x)
+    Expr x_gt_1 = abs(x_full) > 1.0f;
+    if (between_m1_and_p1) {
+        x = x_full;
+    } else {
+        x = select(x_gt_1, 1.0f / x_full, x_full);
+    }
+
+    // Coefficients obtained using src/polynomial_optimizer.py
+    // Note that the maximal errors are computed with numpy with double precision.
+    // The real errors are a bit larger with single-precision floats (see correctness/fast_arctan.cpp).
+
+    // The table is huge, so let's put clang-format off and handle the layout manually:
+    // clang-format off
+    std::vector<float> c;
+    switch (precision) {
+        // == MSE Optimized == //
+        case ApproximationPrecision::MSE_Poly2: // (MSE=1.0264e-05, MAE=9.2149e-03, MaxUlpE=3.9855e+05)
+            c = {+9.762134539879e-01f, -2.000301999499e-01f}; break;
```

Comment:
style nit: please put the `break` on the next line (unless clang-format moves them back onto the same line)

Reply:
Was trying to keep the table as compact as possible. Readability is great IMO.

Reply:
I have put them on the next line.